dspace-server-webapp/src/main/java/org/dspace/app/rest/BitstreamReplaceController.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import static org.dspace.app.rest.utils.RegexUtils.REGEX_REQUESTMAPPING_IDENTIFIER_AS_UUID;
+import static org.springframework.http.MediaType.MULTIPART_FORM_DATA_VALUE;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.UUID;
+
+import jakarta.servlet.http.HttpServletRequest;
+import org.dspace.app.rest.converter.ConverterService;
+import org.dspace.app.rest.model.BitstreamRest;
+import org.dspace.app.rest.model.hateoas.BitstreamResource;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.Bitstream;
+import org.dspace.content.Bundle;
+import org.dspace.content.service.BitstreamService;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.ControllerUtils;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.hateoas.RepresentationModel;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+/**
+ * Controller to replace bitstreams in a bundle.
+ * Endpoint: /api/core/bitstreams/{uuid}/replace
```

Comment:
@ConfusionOrb221 : This new endpoint is undocumented, but it appears to disobey our REST API rules for ["Naming of Endpoints"](https://github.com/DSpace/RestContract/tree/main?tab=readme-ov-file#on-the-naming-of-endpoints) where we say that all "endpoint names MUST use plural nouns instead of verbs" 

So, we cannot use the word `/replace` here as that's a verb.

Couldn't we change this to be a `PUT /api/core/bitstreams/{uuid}/content`?  By definition `PUT` is a replacement, and this endpoint would be replacing just the `/content` while leaving the metadata in tact.